library/std/src/thread/local/tests.rs

```diff
@@ -94,38 +110,54 @@ fn circular() {
         fn drop(&mut self) {
             unsafe {
                 HITS += 1;
-                assert!(K1.try_with(|_| ()).is_ok());
+                assert!(self.0.try_with(|_| ()).is_ok());
                 assert_eq!(HITS, 2);
-                K1.with(|s| *s.get() = Some(S1));
+                self.0.with(|s| *s.get() = Some(S1(self.0, self.1)));
             }
         }
     }
 
     thread::spawn(move || {
-        drop(S1);
+        drop(S1(&K1, &K2));
+    })
+    .join()
+    .unwrap();
+
+    unsafe {
+        HITS = 0;
```

Comment:
This assumes that the TLS destructors have run by the time `join()` returns. This is currently not the case for the x86_64-fortanix-unknown-sgx target.

I'm a bit torn on whether this is the bug in the x86_64-fortanix-unknown-sgx implementation or this assumption is invalid.

Reply:
TLS destructors need to run on the thread that is in the process of exiting. I think `join()` shouldn't return until thread is completely done exiting and thus would have finished running the TLS destructors.

Reply:
I agree, having a thread do *anything* after `join` returned is highly suspicious.

Reply:
See #84409